src/Grids/rectilinear_grid.jl

```diff
@@ -355,19 +355,19 @@ function with_halo(new_halo, old_grid::RectilinearGrid)
     if old_grid.Δxᶠᵃᵃ isa Number
         x = x_domain(old_grid)
     else
-        x = old_grid.xᶠᵃᵃ
+        x = OffsetArray(Array(parent(old_grid.xᶠᵃᵃ)), old_grid.xᶠᵃᵃ.offsets[1])
```

Comment:
```suggestion
        x = xnodes(Face, grid)
```

could also add convenience that a _range_ specification like `x = 0:0.1:10` in the constructor implies constant spacing. Then you can do away with the `if`-statement entirely here.

Reply:
I'm also not sure why this is needed --- can you explain? Why is it not enough to supply the _interior_ nodes to the constructor? I think the grid should be continued outside the domain according to some rule (we shouldn't allow users to set grid spacings outside the domain?)

Reply:
It is just  way to ensure that the faces provided to the constructor are CPU arrays

Reply:
Ok, so perhaps `Array(xnodes(Face, grid))`?